src/views/sign-up-view/index.tsx

```diff
@@ -0,0 +1,120 @@
+// Third-party imports
+import Link from 'next/link'
+
+// HashiCorp imports
+import { IconArrowRight16 } from '@hashicorp/flight-icons/svg-react/arrow-right-16'
+import { IconCheckCircle16 } from '@hashicorp/flight-icons/svg-react/check-circle-16'
+import { IconUserPlus16 } from '@hashicorp/flight-icons/svg-react/user-plus-16'
+import InlineSvg from '@hashicorp/react-inline-svg'
+
+// Global imports
+import useAuthentication from 'hooks/use-authentication'
+import BaseNewLayout from 'layouts/base-new'
+import ErrorView from 'views/error-view-switcher'
+import Button from 'components/button'
+import Heading from 'components/heading'
+import Text from 'components/text'
+
+// Local imports
+import s from './sign-up.module.css'
+
+// Text content
+const TITLE = 'Create Your Account'
+const DETAILS = [
+	'Use your account across HashiCorp products',
+	'Track your progress in longer tutorials',
+	'Bookmark tutorials you need to reference regularly',
+]
+const SIGN_UP_BUTTON_TEXT = 'Sign Up'
+const SIGN_IN_HINT_TEXT = 'Already have an account?'
+const SIGN_IN_BUTTON_TEXT = 'Sign In'
+
+const SignUpView = () => {
+	const { isAuthEnabled, isAuthenticated, isLoading, signIn, signUp } =
+		useAuthentication()
+
+	/**
+	 * @TODO determine loading state UI
+	 */
+	if (isLoading) {
+		return null
+	}
+
+	/**
+	 * Show 404 error if enable_auth is false or the user is already authenticated
+	 */
+	if (!isAuthEnabled || isAuthenticated) {
```

Comment:
💭 I wonder if we could just redirect them back to the homepage instead of showing the 404

Reply:
Good call, I've updated this to have two separate redirects:

- Kept the 404 for `!isAuthEnabled`
- Added a redirect to `/profile` if `isAuthenticated` is true

https://github.com/hashicorp/dev-portal/pull/743/commits/1e7eb3be6e71e29631cd6192f5f84daaed8bf19a